ui/components/app/snaps/snap-ui-renderer/components/footer.ts

```diff
@@ -0,0 +1,32 @@
+import { JSXElement, FooterElement } from '@metamask/snaps-sdk/jsx';
+import { getJsxChildren } from '@metamask/snaps-utils';
+import { NonEmptyArray } from '@metamask/utils';
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+} from '../../../../../helpers/constants/design-system';
+import { mapToTemplate } from '../utils';
+import { UIComponent, UIComponentFactory } from './types';
+
+export const footer: UIComponentFactory<FooterElement> = ({
+  element,
+  ...params
+}) => ({
+  element: 'Box',
+  children: getJsxChildren(element).map((children) =>
+    mapToTemplate({ ...params, element: children as JSXElement }),
+  ) as NonEmptyArray<UIComponent>,
```

Comment:
Since we know Footer only has `SnapFooterButton` as the children, we can handle it here instead of complicating the general button implementation 🤔 

I think that reduces the complexity a bit.